app/models/spree/order.rb

```diff
@@ -673,10 +673,6 @@ def process_each_payment
         break if payment_total >= total
 
         yield payment
-
-        if payment.completed?
-          self.payment_total += payment.amount
-        end
```

Comment:
It's a bit weird to take a payment without updating the total. But I guess that we are just moving the responsibility of updating the total out of the callback. And at some point, we may not want payments in a callback either.

Reply:
The issue was actually cause by this : 
https://github.com/openfoodfoundation/openfoodnetwork/blob/31c717349f9fff7fa15bd027fc934ad165fcf059/app/models/spree/payment.rb#L235-L237 
which is triggered by an `after_save` callback. 
So by the time the code reach these lines the payment was already added to `order.payment_total`. I was hidden by the fact `OrderManagement::Order::Updater#update_totals_and_states` was triggered further down the line, which now does not always happen.

Reply:
I see. I can see that this triggers the same code (`order.payment_total = payments.completed.sum(:amount)`)

If we can be sure that `update_totals_and_states` is always triggered down the line (by a controller), then perhaps we can remove this model callback also! But I'm afraid to suggest it in this PR..

Reply:
I think that would need more investigation. I believe the payment callback is there to ensure the `order.payment_total` is updated when a payment is completed outside the checkout. Ie, capturing a cash payment in the backoffice. There might be other scenarios as well, It's not something I would want to do here.